File: dataset/avatar/Python/Code/codeforces_148_B.py
princess = int ( input ( ) )
dragon = int ( input ( ) )
t = int ( input ( ) )
f = int ( input ( ) )
c = int ( input ( ) )
if dragon <= princess :
    print ( 0 )
    exit ( 0 )
result = 0
position = t * princess
position += princess * position / ( dragon - princess )
while position < c :
    result += 1
    position += princess * position / dragon
    position += princess * f
    position += princess * position / ( dragon - princess )
print ( result )

File: output/deepseek_chat/avatar/algo_based_translation/python/java/codeforces_148_B.java
import java.util.Scanner;

public class codeforces_148_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        double princess = scanner.nextDouble();
        double dragon = scanner.nextDouble();
        double t = scanner.nextDouble();
        double f = scanner.nextDouble();
        double c = scanner.nextDouble();
        
        if (dragon <= princess) {
            System.out.println(0);
            System.exit(0);
        }
        
        int result = 0;
        double position = t * princess;
        position += princess * position / (dragon - princess);
        
        while (position < c) {
            result++;
            position += princess * position / dragon;
            position += princess * f;
            position += princess * position / (dragon - princess);
        }
        
        System.out.println(result);
    }
}
